### PR TITLE
[VidLii] Add 720p support (yt-dlp backport-ish)

### PR DESCRIPTION
**Pull request by mrpapersonic**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [x] New feature

---

### Description of your *pull request* and other information

Adds HD support to VidLii.
This was originally done in #29291, but I deleted that fork a long time ago so I can't make any more edits to it.

### CODE REVIEW

File: youtube_dl/extractor/vidlii.py [hunk 1]
```diff
@@ -46,9 +67,31 @@ def _real_extract(self, url):
...
+                         format_url, "height", default=360))
+
+            formats.append({
+                'url': format_url,
+                'format_id': '%dp' % height if height else None,
+                'height': height,
+            })
+
+        sources = re.findall(
+            r'src\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
```
Comment (dirkf) [inline youtube_dl/extractor/vidlii.py:83]: If the attribute names are all `src` and not `xxxsrc`:
```suggestion
            r'\bsrc\s*:\s*(["\'])(?P<url>(?:https?://)?(?:(?!\1).)+)\1',
```

File: youtube_dl/extractor/vidlii.py [hunk 2]
```diff
@@ -46,9 +67,31 @@ def _real_extract(self, url):
...
+            webpage)
+
+        try:
+            self._request_webpage(sources[1][1], video_id, 'Checking HD URL')
+            add_format(sources[1][1])
+        except ExtractorError:
+            pass
+
+        add_format(sources[0][1])
```
Comment (dirkf) [inline youtube_dl/extractor/vidlii.py:92]: There's a `_check_formats()` method for this:
```suggestion
        formats = []
        if len(sources) > 1:
            formats = [sources[1][1]]
            self._check_formats(formats, video_id)
        if len(sources) > 0:
            formats.append(sources[0][1])
        map(add_format, formats)
```

File: youtube_dl/extractor/vidlii.py [hunk 3]
```diff
@@ -46,9 +67,31 @@ def _real_extract(self, url):
...
+
+        self._sort_formats(formats)
 
         title = self._search_regex(
```
Comment (dirkf) [inline youtube_dl/extractor/vidlii.py:96]: Not part of your PR, but should convert HTML-encoding to text, also strip surrounding spaces:
```suggestion
        title = self._html_search_regex(
```

File: youtube_dl/extractor/vidlii.py [hunk 4]
```diff
@@ -83,9 +126,9 @@ def _real_extract(self, url):
             r'duration\s*:\s*(\d+)', webpage, 'duration', fatal=False))
 
         view_count = int_or_none(self._search_regex(
-            (r'<strong>(\d+)</strong> views',
-             r'Views\s*:\s*<strong>(\d+)</strong>'),
-            webpage, 'view count', fatal=False))
+            (r'<strong>([\d,]+)</strong> views',
+             r'Views\s*:\s*<strong>([\d,]+)</strong>'),
+            webpage, 'view count', fatal=False).replace(",", ""))
```
Comment (dirkf) [inline youtube_dl/extractor/vidlii.py:131]: If these `,`s are thousands separators, might they be `.`s for some locales (plainly not a decimal point for a count)?
```suggestion
        view_count = str_to_int(self._html_search_regex(
            (r'<strong>([\d,.]+)</strong> views',
             r'Views\s*:\s*<strong>([\d,.]+)</strong>'),
            webpage, 'view count', fatal=False))
```
(and `from ..utils import str_to_int` at the top).

**Review by dirkf — CHANGES_REQUESTED** [review]
Thanks for your work!

Just a few observations.

File: youtube_dl/extractor/vidlii.py (hunk 1 above)
Comment (mrpapersonic) [reply youtube_dl/extractor/vidlii.py:84]: The attribute names are `hdsrc` and `src`.

**Comment by mrpapersonic** [comment]
@dirkf can this be re-reviewed? Looks like it got lost in the sea of current PRs.

File: youtube_dl/extractor/vidlii.py [hunk 5]
```diff
@@ -35,6 +38,25 @@ class VidLiiIE(InfoExtractor):
             'categories': ['News & Politics'],
             'tags': ['Vidlii', 'Jan', 'Videogames'],
         }
+    }, {  # HD
```
Comment (dirkf) [inline youtube_dl/extractor/vidlii.py:41]:
```suggestion
    }, {
        # HD
```

**Pull request merged**